File: cpp/n389_FindDiff.cpp
// https://leetcode-cn.com/problems/find-the-difference/
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    char findTheDifference(string s, string t) {
        vector<int> res (26, 0);
        for (char i : t)
            res[i - 'a'] += 1;
        for (char j : s)
            res[j - 'a'] -= 1;
        for (size_t i = 0; i < res.size(); ++i){
            if (res[i] == 1)
                return 'a' + i;
        }
        return -1;
    }
};

int main(){
    string s = "abcd";
    string t = "cedab";
    Solution solu;
    cout << solu.findTheDifference(s, t) << endl;
}

File: python/n389_FindDiff.py
# https://leetcode-cn.com/problems/find-the-difference/

class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        hash = [0] * 26
        for i in t:
            hash[ord(i) - ord('a')] += 1
        for j in s:
            hash[ord(j) - ord('a')] -= 1
        for k in range(len(hash)):
            if hash[k] == 1:
                return chr(97 + k)
                

if __name__ == "__main__":
    solu = Solution()
    s = "abcd"
    t = "dceba"
    print(solu.findTheDifference(s, t))